test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs

```diff
@@ -82,5 +84,124 @@ private static string DownloadIfNotExists(string baseGitPath, string dataFile)
 
             throw new Exception($"Failed to download test file {dataFile}.");
         }
+
+        public static string GetFlowersDataset()
+        {
+            const string datasetName = @"flowers";
+            string assetsRelativePath = @"assets";
+            string assetsPath = GetAbsolutePath(assetsRelativePath);
+            string imagesDownloadFolderPath = Path.Combine(assetsPath, "inputs",
+                "images");
+
+            //Download the image set and unzip
+            string finalImagesFolderName = DownloadImageSet(
+                imagesDownloadFolderPath);
+
+            string fullImagesetFolderPath = Path.Combine(
+                imagesDownloadFolderPath, finalImagesFolderName);
+
+            var images = LoadImagesFromDirectory(
+                folder: fullImagesetFolderPath, useFolderNameAsLabel: true);
+
+            using (StreamWriter file = new StreamWriter(datasetName))
+            {
+                file.WriteLine("Label,ImagePath");
+                foreach (var image in images)
+                    file.WriteLine(image.Label + "," + image.ImagePath);
+            }
+
+            return datasetName;
+        }
+
+        public static IEnumerable<ImageData> LoadImagesFromDirectory(string folder,
```

Comment:
This method is already duplicated twice in our tests. It is time to extract it to a common helper method.

Reply:
I would ideally but because I'm short on time I can open an issue and extract later.

---
In reply to: [339654988](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339654988) [](ancestors = 339654988)

Reply:
Can you add a link here to the issue?